### PR TITLE
Ports: Split up the `halflife` port into engine and game

### PR DESCRIPTION
**Pull request by timschumi**
This keeps us from having to do gymnastics regarding the patches since we are otherwise trying to patch and build two separate tarballs at the same time.

### CODE REVIEW

**Comment by EWouters** [comment]
I don't have a copy of halflife handy but it looks fine and compiles. Nice to see this yak shaved!

File: Ports/xash3d-fwgs/package.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+depends=("SDL2" "fontconfig" "freetype")
+workdir="xash3d-fwgs-${_fwgs_commit}"
+files="https://github.com/FWGS/xash3d-fwgs/archive/${_fwgs_commit}.tar.gz xash3d-fwgs-${_fwgs_commit}.tar.gz 1401f6c0cf619c48a8a40938b2acdffd327725ca0ab59804c518bddf821637f9
+https://github.com/FWGS/vgui-dev/archive/${_vgui_commit}.tar.gz vgui-dev-${_vgui_commit}.tar.gz eb9315fba8ae444fdae240c10afebaf7f3b157233bf1589f0af557b2286928fa
+https://github.com/FWGS/mainui_cpp/archive/${_mainui_commit}.tar.gz mainui_cpp-${_mainui_commit}.tar.gz c8f6ce81596d5690044542074ac9bc69bbd43b5e5766f71363a8b5d4d382ad71
+https://github.com/FWGS/MiniUTL/archive/${_miniutl_commit}.tar.gz MiniUTL-${_miniutl_commit}.tar.gz 7b7b26377854b3fc741c8d652d8b3c9c540512644943ca6efb63df941b2861e3"
+auth_type=sha256
+launcher_name="Half-Life"
+launcher_category="Games"
+launcher_command="sh /home/anon/Games/halflife/hl.sh"
```
Comment (linusg) [inline Ports/xash3d-fwgs/package.sh:18]: Why is this duplicated from `halflife`? I don't think installing the engine should create a launcher

Comment (timschumi) [reply Ports/xash3d-fwgs/package.sh]: `xash3d-fwgs` started out as a copy of the `halflife` script, after which I removed all the parts that don't apply for the respective package. Looks like I missed a few lines.

Should be fixed now.

Comment (linusg) [reply Ports/xash3d-fwgs/package.sh]: Got it, thanks!

**Pull request merged**